OneCommunityGlobal/HighestGoodNetworkApp
# Wanting new make 5 letter codes editable only by admin&owner

# Description
<img width="853" alt="截屏2023-11-03 上午11 16 18" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/d70a705c-9b37-4b22-84a6-76df1eb872f2">



## Related PRS (if any):
BE # **[648](https://github.com/OneCommunityGlobal/HGNRest/pull/648)**

## Main changes explained:
- Update WeeklySummariesReport.jsx file for Owner/Admin/whoever is given "edit team code" permission to edit 5-letter team code in 'Weekly Summaries Report page'
- Update UserProfile.jsx file for Owner/Admin/whoever is given "edit team code" permission to edit 5-letter team code in profile's teams tab
- Updated other related files

## How to test:
1. check into current branch and BE # **[648](https://github.com/OneCommunityGlobal/HGNRest/pull/648)**
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as owner/admin user
        a. go to Reports -> Weekly Summaries Report
        b. verify that the 5-letter team code is editable
        c. go to User Profile -> Teams tab
        d. verify that the 5-letter team code is editable
4. edit a team code, refresh the page, and check the edited team code is saved
5. log as any role EXCEPT admin/owner(like volunteer, manager, mentor, etc.)
        a. go to Reports -> Weekly Summaries Report
        b. verify that the 5-letter team code is NOT editable
        c. go to User Profile -> Teams tab
        d. verify that the 5-letter team code is NOT editable
6. log as owner, go to "Other Links" -> Permissions Management -> Manage User Permissions -> search for the user who doesn't have permission to edit team code -> select "See Only Weekly Summary Reports Tab"(at very front) and "Edit Team 4-Digit Codes"(at the end) and submit
7. log as the user in step 6, go to Reports -> Weekly Summaries Report and User Profile -> Teams tab, verify that the 5-letter team code is editable


## Screenshots or videos of changes:
**editable team code example:**
![截屏2023-10-20 下午1 11 24](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/e594c116-42a4-4e8f-b9d1-7914b831fe6d)

![截屏2023-10-20 下午1 45 17](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/10a1ff4d-5f99-4853-b033-03b710626976)


**uneditable team code example:**
![截屏2023-10-20 下午1 17 32](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/eb455ec6-05f4-4859-a67f-c4d45c443c33)

![截屏2023-10-20 下午1 21 32](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/4a4e3de3-9958-41d2-82a9-ac63af0d85bc)
